### PR TITLE
Fix 8 python packages in python-unstable

### PR DESCRIPTION
**Pull request by costrouc**
###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Review by teto — APPROVED** [review]
Some of the edits seem like unnecessary churn unless we have a specific nix style to respect (the conversion from one liners to one package per line for instance) but overall looks good.

File: pkgs/development/python-modules/jsonschema/default.nix
Comment (teto) [inline pkgs/development/python-modules/jsonschema/default.nix]: is @domenkozar ok with the change ?

File: pkgs/development/python-modules/jsonschema/default.nix
Comment (worldofpeace) [inline pkgs/development/python-modules/jsonschema/default.nix]: Is this not native?

File: pkgs/development/python-modules/jsonschema/default.nix
Comment (worldofpeace) [inline pkgs/development/python-modules/jsonschema/default.nix]: I think @FRidh and the lot converged on python packages using the base32 hash

File: pkgs/development/python-modules/glob2/default.nix
Comment (worldofpeace) [inline pkgs/development/python-modules/glob2/default.nix]:
```suggestion
    homepage = https://github.com/miracle2k/python-glob2/;
```

https

File: pkgs/development/python-modules/jsonschema/default.nix
Comment (costrouc) [reply pkgs/development/python-modules/jsonschema/default.nix]: Whoops that was not intentional. I have a tool that auto generates the derivations and I forgot to add him back in. I will add him back

File: pkgs/development/python-modules/glob2/default.nix
Comment (worldofpeace) [inline pkgs/development/python-modules/glob2/default.nix]: Should be base32

File: pkgs/development/python-modules/pytest-annotate/default.nix
Comment (worldofpeace) [inline pkgs/development/python-modules/pytest-annotate/default.nix]:
```suggestion
  postPatch = ''
```

File: pkgs/development/python-modules/pytest-annotate/default.nix
Comment (worldofpeace) [inline pkgs/development/python-modules/pytest-annotate/default.nix]: I've been suggesting making this a `buildInputs` so we can allow pytest version mixing.

##### not sure what you think of that

File: pkgs/development/python-modules/pytest-mock/default.nix
Comment (worldofpeace) [inline pkgs/development/python-modules/pytest-mock/default.nix]: This is not native?

**Review by worldofpeace — CHANGES_REQUESTED** [review]
Can you drop
* pythonPackages.pydot_ng: refactor fix broken build

#64462 is a more complete fix, but I'm not sure if it should be removed from nixpkgs
since it's dead https://github.com/NixOS/nixpkgs/pull/64462#issuecomment-510563072

File: pkgs/development/python-modules/pytest-annotate/default.nix
Comment (FRidh) [reply pkgs/development/python-modules/pytest-annotate/default.nix]: Correct. We should document this somewhere.

File: pkgs/development/python-modules/pytest-mock/default.nix
Comment (FRidh) [reply pkgs/development/python-modules/pytest-mock/default.nix]: Should be

File: pkgs/development/python-modules/glob2/default.nix
Comment (FRidh) [reply pkgs/development/python-modules/glob2/default.nix]: Note we have not documented this yet. I am not doing it yet either because it's always a bit of a hassle even though I agree we should go for base32.
https://github.com/ryantm/nixpkgs-update/issues/106

File: pkgs/development/python-modules/jsonschema/default.nix
Comment (costrouc) [reply pkgs/development/python-modules/jsonschema/default.nix]: Fixed.

File: pkgs/development/python-modules/jsonschema/default.nix
Comment (costrouc) [reply pkgs/development/python-modules/jsonschema/default.nix]: Fixed.

File: pkgs/development/python-modules/jsonschema/default.nix
Comment (costrouc) [reply pkgs/development/python-modules/jsonschema/default.nix]: Fixed.

File: pkgs/development/python-modules/glob2/default.nix
Comment (costrouc) [reply pkgs/development/python-modules/glob2/default.nix]: Fixed.

File: pkgs/development/python-modules/glob2/default.nix
Comment (costrouc) [reply pkgs/development/python-modules/glob2/default.nix]: Fixed.

File: pkgs/development/python-modules/pytest-mock/default.nix
Comment (costrouc) [reply pkgs/development/python-modules/pytest-mock/default.nix]: Fixed.

File: pkgs/development/python-modules/pytest-annotate/default.nix
Comment (costrouc) [reply pkgs/development/python-modules/pytest-annotate/default.nix]: Fixed.

File: pkgs/development/python-modules/pytest-annotate/default.nix
Comment (costrouc) [reply pkgs/development/python-modules/pytest-annotate/default.nix]: Fixed.

**Comment by costrouc** [comment]
@worldofpeace I have removed `pythonPackages.pydot_ng` from the PR. Ready for merge. I have nix-review running and everything looks good so far (about 50% done)

**Comment by worldofpeace** [comment]
@FRidh I feel like the situation with the hashes being base32 probably isn't something to document specifically for python packages, but perhaps in general.

As for about pytest, I'm not sure if we've documented the situation with not mixing versions globally in
`9.14.4.1. Contributing guidelines` but it would be easy to add a note about pytest packages if something like that was there.

**Comment by worldofpeace** [comment]
@costrouc When you supply the fixes for the requested changes it would help reviewers if you did easily fixup'able commits so the changes can easily be seen on github. That way I don't have to start over with the review, even if we're certain we got everything.

Otherwise LGTM, be sure to paste your nix-review report in a gist.

**Comment by FRidh** [comment]
@worldofpeace you mean contrary to fixing up the commits directly? I always request for that instead, so I can just merge when I need to, instead of asking yet another time for squashing them. I guess there's pros and cons to both.

**Comment by worldofpeace** [comment]
@FRidh That's because it's convenient for me to just fixup the authors history real quick and merge once I know it's good.

**Comment by costrouc** [comment]
@worldofpeace so I should create a fixup commit after all the packages? Right now I do some rebase/fixup trickery so that I can keep each commit separate and dealing with a single package each.

**Comment by worldofpeace** [comment]
@costrouc Essentially

Original commit
```
pythonPackages.jsonschema: 2.6.0 -> 3.0.1
```

"Please drop x input not needed" on jsonschema

```
pythonPackages.jsonschema: drop x input
```

Or even just a fixup! commit.
Is it uncommon for people to do this?

**Comment by costrouc** [comment]
If I follow you are suggesting that I create additional commits and you will handle combining commits once the PR is ready for merge. I will do that from now on if there are any more recommended changes.

**Comment by worldofpeace** [comment]
Yeah essentially, I think @FRidh is suggesting that it just adds an extra step for merge since you're waiting for the author to fix the history.

**Comment by teto** [comment]
As @worldofpeace, I prefer fixup commits  but I don't think it's possible from github's ui to squash the fixup ? does the rebase do that ?

**Comment by worldofpeace** [comment]
@teto Oh no, I checkout the pr locally and just force push that.

**Comment by costrouc** [comment]
Ready for merge. Here are the logs of nix-review https://gist.github.com/costrouc/b5bd51c2736378c0e5a3c5831e085fbd

I didn't wait for the nix-review to finish because I've been waiting on the last 5 packages for about 2 hours

**Comment by costrouc** [comment]
@worldofpeace of peace is it normal for so many packages to not build in a nix-review?

**Comment by worldofpeace** [comment]
@costrouc Some of the failures could have been caused by https://github.com/NixOS/nixpkgs/pull/64952/commits/bf540792304835d573db27b4cc43e326ed5ffd46.

But we should be able to use https://hydra.nixos.org/jobset/nixpkgs/python-unstable to figure this out.
I think we're good for merge.

**Comment by costrouc** [comment]
This broke aws-sam-translator which many packages depend on https://hydra.nixos.org/eval/1530873

**Comment by worldofpeace** [comment]
I believe this release fixes that
* https://github.com/awslabs/serverless-application-model/releases/tag/v1.12.0

**Comment by costrouc** [comment]
I'll submit a PR for that

**Pull request merged**
